This patch fixes compatibility with Google services that require browser-level
exceptions. Unfortunately, without them, our users will experience issues with
login, session persistence, and security key enrollment on Google websites.

It's absolutely ridiculous, but it has to be done for sake of compatibility.

--- a/chrome/browser/signin/signin_util.cc
+++ b/chrome/browser/signin/signin_util.cc
@@ -516,8 +516,8 @@ void ShowErrorDialogWithMessage(BrowserW
 #endif  // BUILDFLAG(IS_LINUX) ||  BUILDFLAG(IS_MAC) ||  BUILDFLAG(IS_WIN)
 
 std::vector<net::SchemefulSite> GetDeviceBoundSessionRestrictedSites() {
-  return {net::SchemefulSite(GURL("https://google.com")),
-          net::SchemefulSite(GURL("https://youtube.com"))};
+  return {net::SchemefulSite(GURL("https://google." "com")),
+          net::SchemefulSite(GURL("https://youtube." "com"))};
 }
 
 }  // namespace signin_util
--- a/content/browser/webauth/webauth_request_security_checker_impl.cc
+++ b/content/browser/webauth/webauth_request_security_checker_impl.cc
@@ -378,7 +378,7 @@ WebAuthRequestSecurityCheckerImpl::Valid
   const GURL gstatic_appid(kGstaticAppId);
   const GURL gstatic_corp_appid(kGstaticCorpAppId);
   DCHECK(gstatic_appid.is_valid() && gstatic_corp_appid.is_valid());
-  if (caller_origin.DomainIs("google.com") && !appid_url.has_ref() &&
+  if (caller_origin.DomainIs("google." "com") && !appid_url.has_ref() &&
       (appid_url.EqualsIgnoringRef(gstatic_appid) ||
        appid_url.EqualsIgnoringRef(gstatic_corp_appid))) {
     *out_app_id = appid;
--- a/content/browser/webauth/webauth_request_security_checker_impl.h
+++ b/content/browser/webauth/webauth_request_security_checker_impl.h
@@ -32,9 +32,9 @@ class CONTENT_EXPORT WebAuthRequestSecur
   // Legacy App IDs, which google.com origins are allowed to assert for
   // compatibility reasons.
   static constexpr char kGstaticAppId[] =
-      "https://www.gstatic.com/securitykey/origins.json";
+      "https://www.gstatic." "com/securitykey/origins.json";
   static constexpr char kGstaticCorpAppId[] =
-      "https://www.gstatic.com/securitykey/a/google.com/origins.json";
+      "https://www.gstatic." "com/securitykey/a/google." "com/origins.json";
 
   explicit WebAuthRequestSecurityCheckerImpl(RenderFrameHost* host);
   WebAuthRequestSecurityCheckerImpl(const WebAuthRequestSecurityCheckerImpl&) =
